lib/dry/web/roda/generators/abstract_project.rb

```diff
@@ -0,0 +1,52 @@
+require "dry/web/roda/generate"
+
+module Dry
+  module Web
+    module Roda
+      module Generators
+        class AbstractProject
+          attr_reader :target_dir, :options
+
+          def initialize(target_dir, options = {})
+            @target_dir = target_dir
+            @options = options
+            @generator = Generate.new(destination)
+            @templates = []
+          end
+
+
+          def call
+            map_templates.each do |source, target|
+              @generator.(source, target, prepare_scope)
```

Comment:
Since the output of `prepare_scope` should be the same every time, perhaps we can pass it as an arg to `Generate.new`? That way we won't need to to keep regenerating it and passing it again with every file.